src/web/services/student.service.spec.ts

```diff
@@ -75,25 +73,151 @@ describe('StudentService', () => {
       courseid: 'CS3281',
       studentemail: 'johndoe@gmail.com',
     };
+    spyOn(spyHttpRequestService, 'post').and.stub();
 
     service.regenerateStudentCourseLinks(paramMap.courseid, paramMap.studentemail);
 
     expect(spyHttpRequestService.post)
         .toHaveBeenCalledWith(ResourceEndpoints.STUDENT_COURSE_LINKS_REGENERATION, paramMap);
   });
 
-  it('should execute GET when loading students in a course as CSV', () => {
-    const paramMap: Record<string, string> = {
-      courseid: 'CS3281',
-    };
+  it('should execute GET on course & students endpoints when generating student csv list', () => {
+    const courseId: string = 'CS3281';
+    const httpSpy: Spy = spyOn(spyHttpRequestService, 'get').and.returnValue(of(''));
+    const studentList: Observable<string> = service.loadStudentListAsCsv({ courseId });
+    expect(httpSpy).toBeCalledWith(ResourceEndpoints.COURSE, {
+      courseid: courseId,
+      entitytype: 'instructor',
+    });
+    studentList.subscribe(() => expect(httpSpy).toBeCalledWith(ResourceEndpoints.STUDENTS, {
+      courseid: courseId,
+    }));
+  });
 
-    const responseType: string = 'text';
+  it('should show course student list with section', () => {
```

Comment:
The description should mention about csv as `it` here refers to the service itself.